bitnami/openldap/2.6/debian-12/rootfs/opt/bitnami/scripts/libopenldap.sh

```diff
@@ -226,6 +226,31 @@ is_ldap_not_running() {
     ! is_ldap_running
 }
 
+#########################
+# Wait for OpenLDAP to become ready
+# Arguments:
+#   None
+# Returns:
+#   Whether slapd is ready for queries
+########################
+is_ldap_ready() {
+    local attempts=5
+    local delay=5
+
+    info "Waiting for LDAP to be ready"
+    for ((i=1; i<=attempts; i++)); do
+        sleep "$delay"
+
+        if debug_execute ldapsearch -Y EXTERNAL -H "ldapi:///" -b "cn=config" > /dev/null 2>&1; then
+            info "LDAP is ready."
+            return 0
+        fi
+        done
```

Comment:
Let's change this snippet of code with a `retry_while` construction to be consistent with the rest of the code base. I understand something like the snippet below should do the trick

```
if ! retry_while "debug_execute ldapsearch -Y EXTERNAL -H 'ldapi:///' -b 'cn=config'" "$retries" "$sleep_time"; then
    error "OpenLDAP failed to start in the background."
    return 1
else
    info "OpenLDAP is ready"
fi
```